packages/hash/frontend/src/shared/layout/layout-with-sidebar/account-page-list/account-page-list.tsx

```diff
@@ -0,0 +1,284 @@
+import { NavLink } from "../nav-link";
+import { AccountPageListItem } from "./account-page-list-item";
+import { useReorderPage } from "../../../../components/hooks/useReorderPage";
+import { TreeElement, getProjection, getPageList } from "./utils";
+
+type AccountPageListProps = {
+  accountId: string;
+  currentPageEntityId?: string;
+};
+
+const measuringConfig = {
+  droppable: {
+    strategy: MeasuringStrategy.Always,
+  },
+};
+
+export const AccountPageList: FunctionComponent<AccountPageListProps> = ({
+  currentPageEntityId,
+  accountId,
+}) => {
+  const { data } = useAccountPages(accountId);
+
+  const { createUntitledPage, loading: createUntitledPageLoading } =
+    useCreatePage(accountId);
+  const { createSubPage, loading: createSubpageLoading } =
+    useCreateSubPage(accountId);
+  const { reorderPage, loading: reorderLoading } = useReorderPage(accountId);
+  const { archivePage, loading: archivePageLoading } = useArchivePage();
+
+  const [expandedPageIds, setExpandedPageIds] = useLocalstorageState<string[]>(
+    "hash-expanded-sidebar-pages",
+    [],
+  );
+
+  const [pagesTreeList, setPagesTreeList] = useState<TreeElement[]>([]);
+  const [pagesFlatList, setPagesFlatList] = useState<TreeElement[]>([]);
+  const [pagesFlatIdList, setPagesFlatIdList] = useState<string[]>([]);
+
+  const [activeId, setActiveId] = useState<UniqueIdentifier | null>(null);
+  const [overId, setOverId] = useState<UniqueIdentifier | null>(null);
+  const [offsetLeft, setOffsetLeft] = useState(0);
+
+  const loading =
+    createUntitledPageLoading ||
+    createSubpageLoading ||
+    reorderLoading ||
+    archivePageLoading;
+
+  // @todo handle loading/error states properly
+  const addPage = useCallback(async () => {
+    if (loading) {
+      return;
+    }
+
+    try {
+      await createUntitledPage();
+    } catch (err) {
+      // eslint-disable-next-line no-console -- TODO: consider using logger
+      console.error("Could not create page: ", err);
+    }
+  }, [createUntitledPage, loading]);
+
+  const handleToggle = (nodeId: string) => {
+    setExpandedPageIds((expandedIds) =>
+      expandedIds.includes(nodeId)
+        ? expandedIds.filter((id) => id !== nodeId)
+        : [...expandedIds, nodeId],
+    );
+  };
+
+  const setSortableList = (
+    list: AccountPage[],
+    expandedPageIdList: string[],
+  ) => {
+    const { treeList, flatList } = getPageList(list, expandedPageIdList);
+    setPagesTreeList(treeList);
+    setPagesFlatList(flatList);
+    setPagesFlatIdList(flatList.map((page) => page.entityId));
+  };
+
+  useMemo(() => {
+    if (!loading) {
+      setSortableList(data, expandedPageIds);
+    }
+  }, [data, expandedPageIds, loading]);
+
+  const sensors = useSensors(
+    useSensor(PointerSensor, {
+      activationConstraint: {
+        distance: 5,
+      },
+    }),
+  );
+
+  const projected =
+    activeId && overId
+      ? getProjection(pagesFlatList, activeId, overId, offsetLeft, 16)
```

Comment:
`16` here is a magic number but I think is supposed to match something in styles? Can this be a variable?

Reply:
Moved it to a variable in c762b94d0bdc79fd988f6ed3b7364ed3f31e24a9